src/libsyncengine/jobs/network/upload_session/uploadsession.cpp

```diff
@@ -285,62 +285,72 @@ bool UploadSession::sendChunks() {
-                    jobCreationError = true;
-                    break;
-                }
+            }
 
-                if (XXH3_64bits_update(state, chunkJob->chunkHash().data(), chunkJob->chunkHash().length()) == XXH_ERROR) {
-                    LOGW_WARN(_logger,
-                              L"Checksum computation " << jobId() << L" failed for file " << Path2WStr(_filePath).c_str());
-                    checksumError = true;
-                    break;
-                }
+            std::streamsize actualChunkSize = file.gcount();
+            std::string chunkContent = std::string(memblock.get(), actualChunkSize);
 
-                if (_liteSyncActivated) {
-                    // Set VFS callbacks
-                    chunkJob->setVfsUpdateFetchStatusCallback(_vfsUpdateFetchStatus);
-                    chunkJob->setVfsSetPinStateCallback(_vfsSetPinState);
-                    chunkJob->setVfsForceStatusCallback(_vfsForceStatus);
-                }
+            if (actualChunkSize <= 0) {
+                LOG_ERROR(_logger, "Chunk size is 0");
+#ifdef NDEBUG
+                sentry_capture_event(
+                    sentry_value_new_message_event(SENTRY_LEVEL_WARNING, "Upload chunke error", "Chunk size is 0"));
+#endif
+                readError = true;
+                break;
+            }
 
-                if (_isAsynchrounous) {
-                    std::function<void(UniqueId)> callback =
-                        std::bind(&UploadSession::uploadChunkCallback, this, std::placeholders::_1);
-
-                    _mutex.lock();
-                    _threadCounter++;
-                    JobManager::instance()->queueAsyncJob(chunkJob, Poco::Thread::PRIO_NORMAL, callback);
-                    _ongoingChunkJobs.insert({chunkJob->jobId(), chunkJob});
-                    _mutex.unlock();
-                    LOG_INFO(_logger, "Session " << _sessionToken.c_str() << ", job " << chunkJob->jobId() << " queued, "
-                                                 << _threadCounter << " jobs in queue");
-
-                    waitForJobsToComplete(false);
-                } else {
-                    LOG_INFO(_logger, "Session " << _sessionToken.c_str() << ", thread " << chunkJob->jobId() << " start.");
-
-                    ExitCode exitCode = chunkJob->runSynchronously();
-                    if (exitCode != ExitCodeOk || chunkJob->hasHttpError()) {
-                        LOGW_WARN(_logger, L"Failed to upload chunk " << chunkNb << L" of file "
-                                                                      << Path2WStr(_filePath.filename()).c_str());
-                        _exitCode = exitCode;
-                        _exitCause = chunkJob->exitCause();
-                        _jobExecutionError = true;
-                        break;
-                    }
-
-                    _progress += actualChunkSize;
+            std::shared_ptr<UploadSessionChunkJob> chunkJob = nullptr;
```

Comment:
It seems that a shared pointer is initialized to `nullptr` by default: https://stackoverflow.com/questions/57310392/will-a-default-constructed-empty-shared-ptr-automatically-be-initialized-to-nu.